### PR TITLE
radarrv3: check and wait for the api to be alive with a 30s timeout a…

### PR DESCRIPTION
**Pull request by userdocs**
## Proposed Changes:
- Test for the api to be responsive instead of sleep and hope

## Tick these if they apply
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Some existing `utils` got changed

## Checklist
- [x] Docs have been made/are not necessary
    - PR link: 
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [x] I have commented my code, particularly in hard-to-understand areas

## Testing done
OS & Version:

### How have you tested this

Scenarios tested:
- I ran this and it worked
- I ran that and it didn't work

### CODE REVIEW

File: scripts/update/radarr.sh [hunk 1]
```diff
@@ -1,77 +1,78 @@
...
-			echo_progress_start "Downloading source files"
-			if ! curl "https://radarr.servarr.com/v1/update/master/updatefile?os=linux&runtime=netcore&arch=x64" -L -o /tmp/Radarr.tar.gz >> "$log" 2>&1; then
-				echo_error "Download failed, exiting"
+. /etc/swizzin/sources/functions/radarr
+if [[ $(_radarr_version) != 'mono-v2' ]]; then
+	if [[ -f /install/.radarr.lock ]]; then
+		#Move v3mono installs to v3.net
+		if grep -q "ExecStart=/usr/bin/mono" /etc/systemd/system/radarr.service; then
+			echo_info "Moving Radarr from mono to .Net"
+			echo_log_only "Found radarr service pointing to mono"
```
Comment (flying-sausages) [inline scripts/update/radarr.sh:8]:
```suggestion

```
Not necessary seeing as echo_info pushes to log actually, silly me

File: scripts/update/radarr.sh [hunk 2]
```diff
@@ -1,77 +1,78 @@
...
-			echo_progress_done "Source downloaded"
 
-			echo_progress_start "Extracting archive"
-			systemctl stop radarr -q
-			rm /opt/Radarr/Radarr.exe
-			tar -xvf /tmp/Radarr.tar.gz -C /opt >> "$log" 2>&1
-			chown -R "$radarrOwner":"$radarrOwner" /opt/Radarr
-			echo_progress_done "Archive extracted"
+			##TODO find a different way to check this seeing as we need to query Radarr API, would ben nicer to do from FS
+			if [[ $isnetcore = "false" ]]; then # This case confirms we are running on v3 without .net core, i.e. the case we want to update
```
Comment (flying-sausages) [inline scripts/update/radarr.sh:35]: Is there a better way to determine if we're on mono now, other than querying the API?

File: sources/functions/radarr [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+	if radarrv="$(/usr/bin/mono /opt/Radarr/Radarr.exe \? | head -n 1 | grep -Eo "Version(.*)$")"; then
+		[[ "$radarrv" =~ ^(Version 0.2.(.*))$ ]] && echo "mono-v2"
+		[[ "$radarrv" =~ ^(Version 3.0.(.*))$ ]] && echo "mono-v3"
+	elif radarrv="$(/opt/Radarr/Radarr \? | head -n 1 | grep -Eo "Version(.*)$")"; then
+		[[ "$radarrv" =~ ^(Version 3.0.(.*))$ ]] && echo "netcore-v3"
+	else
+		echo "there is no spoon?"
+	fi
+}
+export -f _radarr_version
```
Comment (flying-sausages) [inline sources/functions/radarr:13]:
```suggestion
```
Not necessary seeing as this won't get sourced in box, and can be unloaded when the script where it was sourced dies

File: sources/functions/radarr [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+_radarr_version() {
+	if radarrv="$(/usr/bin/mono /opt/Radarr/Radarr.exe \? | head -n 1 | grep -Eo "Version(.*)$")"; then
+		[[ "$radarrv" =~ ^(Version 0.2.(.*))$ ]] && echo "mono-v2"
+		[[ "$radarrv" =~ ^(Version 3.0.(.*))$ ]] && echo "mono-v3"
+	elif radarrv="$(/opt/Radarr/Radarr \? | head -n 1 | grep -Eo "Version(.*)$")"; then
+		[[ "$radarrv" =~ ^(Version 3.0.(.*))$ ]] && echo "netcore-v3"
+	else
+		echo "there is no spoon?"
```
Comment (flying-sausages) [inline sources/functions/radarr:10]:
```suggestion
		echo_error "there is no spoon?"
```
would an `exit 1` also be worthy here?

**Review by flying-sausages — CHANGES_REQUESTED** [review]

File: scripts/update/radarr.sh (hunk 2 above)
Comment (userdocs) [reply scripts/update/radarr.sh:34]: we can use the function, if that is all we need to know?

File: sources/functions/radarr (hunk 4 above)
Comment (userdocs) [reply sources/functions/radarr:10]: yes, but not an exit. maybe a log or warning

File: scripts/update/radarr.sh [hunk 5]
```diff
@@ -1,77 +1,78 @@
...
-
-		##TODO find a different way to check this seeing as we need to query Radarr API, would ben nicer to do from FS
-		if [[ $isnetcore = "false" ]]; then # This case confirms we are running on v3 without .net core, i.e. the case we want to update
-
-			echo_progress_start "Downloading source files"
-			if ! curl "https://radarr.servarr.com/v1/update/master/updatefile?os=linux&runtime=netcore&arch=x64" -L -o /tmp/Radarr.tar.gz >> "$log" 2>&1; then
-				echo_error "Download failed, exiting"
+. /etc/swizzin/sources/functions/radarr
+if [[ $(_radarr_version) != 'mono-v2' ]]; then
+	if [[ -f /install/.radarr.lock ]]; then
```
[the remark this answers is not in the document]
Comment (userdocs) [reply scripts/update/radarr.sh:4]: i did this to test. test it!!!

[the remark this answers is not in the document]
Comment (userdocs) [reply scripts/update/radarr.sh]: done

File: scripts/update/radarr.sh [hunk 6]
```diff
@@ -25,14 +24,8 @@ if [[ $(_radarr_version) != 'mono-v2' ]]; then
 			ret=$(curl -sL "http://127.0.0.1:7878/api/v3/system/status?apiKey=${apikey}")
 			#
 			echo_log_only "Content of ret =\n ${ret}"
```
Comment (flying-sausages) [inline scripts/update/radarr.sh]:
```suggestion
```
As far as I can see `$ret` is not useful anymore if we don't have to use the API

**Review by flying-sausages — CHANGES_REQUESTED** [review]

**Review by flying-sausages — APPROVED** [review]

File: scripts/nginx/radarr.sh [hunk 7]
```diff
@@ -43,26 +43,25 @@ cat > /home/"$user"/.config/Radarr/config.xml << SONN
...
 chown -R "$user":"$user" /home/"$user"/.config/Radarr
 
 #shellcheck source=sources/functions/utils
 . /etc/swizzin/sources/functions/utils
 systemctl start radarr -q # Switch radarr on regardless whether it was off before or not as we need to have it online to trigger this cahnge
+if ! timeout 15 bash -c -- "while ! curl -fL \"http://127.0.0.1:7878/api/v3/system/status?apiKey=${apikey}\" >> \"$log\" 2>&1; do sleep 5; done"; then
+	echo_error "Radarr API did not respond as expected. Please make sure Radarr is on v3 and running."
+	exit 1
+else
+	urlbase="$(curl -sL "http://127.0.0.1:7878/api/v3/config/host?apikey=${apikey}" | jq '.urlBase' | cut -d '"' -f 2)"
```
Comment (flying-sausages) [inline scripts/nginx/radarr.sh:57]: We don't have to use the API for this, but it works so w/e

**Pull request merged**